src/database/include/database/context.hh

```diff
@@ -0,0 +1,63 @@
+#pragma once
```

Comment:
In general, there is no need to have separated directories for `include` and `source` but if you want to keep files in this way there is also no problem.